src/libServer/LookupServer.cpp

```diff
@@ -1067,6 +1067,34 @@ Json::Value LookupServer::CreateTransactionEth(
   }
 }
 
+TxBlock LookupServer::GetBlockByTransactionHash(const std::string& txnhash) {
+  // Scan downwards looking for the block hash with our TX in it
+  const auto txBlock = m_mediator.m_txBlockChain.GetLastBlock();
+  auto height = txBlock.GetHeader().GetBlockNum() ==
+                    std::numeric_limits<uint64_t>::max()
+                    ? 1
+                    : txBlock.GetHeader().GetBlockNum();
+
+  TxnHash argHash{txnhash};
+  // Scan downwards through the chain until the TX can be found
+  do {
+    auto const block = GetEthBlockByNumber(std::to_string(height), false);
+    height--;
+
+    // Attempt to find if the TX is within this block
+    auto const TxnHashes = block["transactions"];
+    for (auto const& item : TxnHashes) {
+      TxnHash hash_1{item.asString()};
+      if (hash_1 == argHash) {
+        return block;
+      }
+    }
+  } while (height != 0);
```

Comment:
There's also a problem of going back too far, but I guess it can stop once GetEthBlockByNumber returns null